src/Pirouette/Symbolic/Eval.hs

```diff
@@ -125,16 +128,16 @@ instance MonadFail (SymEval lang) where
   fail = error
 
 symeval ::
-  (SymEvalConstr lang, PirouetteDepOrder lang m) =>
+  (SymEvalConstr lang, PirouetteDepOrder lang m, MonadIO m) =>
   Options ->
   SymEval lang a ->
   m [Path lang a]
 symeval opts prob = do
   defs <- getPrtOrderedDefs
-  return $ runSymEval opts defs def prob
+  liftIO $ runSymEval opts defs def prob
 
 symevalAnyPathAccum ::
-  (SymEvalConstr lang, PirouetteDepOrder lang m) =>
+  (SymEvalConstr lang, PirouetteDepOrder lang m, MonadIO m) =>
```

Comment:
`runSymEvalWorker` (called in the body of this function now returns an `IO` computation so we need the resulting monadic type `m` to be an instance of `MonadIO`.